### ISSUE TITLE
suggest consider using http://obofoundry.org as a model for creating registry

### ISSUE THREAD

**Issue by jhpoelen** [open-traits-network/TraitRegistry#1]
Hi!

Am really excited about the trait registry taking shape!

In case you are not familiar with http://obofoundry.org , I'd highly recommend to consider or perhaps adopt the way they've implemented their registry of ontologies: rather than using a database of sorts, they leverage commonly use static site generation (jekyll) along with "front-matter" to generate pretty neat pages via https://github.com/OBOFoundry/OBOFoundry.github.io without even having to run a server themselves or worry about database administration.

So benefits include:
1. no need to run server / maintain database
2. uses git/github to do version control on the registry entries
3. pull requests can be used for others to contribute
4. site is easy to run locally (or anywhere really) using jekyll

Hope this helps,
-jorrit

**Comment by jhpoelen** [comment]
Here's a link to the files that capture descriptions and meta-data of ontology entries:
https://github.com/OBOFoundry/OBOFoundry.github.io/tree/master/ontology

**Comment by bmaitner** [comment]
I really like the idea of doing as much as possible via Github, and the use of markdown files should be easy enough for most folks to populate from a template.  I would like for folks that aren't comfortable using Github to be able to contribute as well, so perhaps having a form option as well could be useful.  Then we simply need to write a script that goes through the data submitted via the form and uses it to generate .md documents.

**Comment by caterinap** [comment]
I also think that having everything on Github would be ideal. And @bmaitner point about having an alternative for users that are not comfortable with Git is a very good one. @jmadin is working on the programming part (and this is a bit out of my comfort zone). I guess that we can wait for his beta version of the registry and start from there.

**Comment by iimog** [comment]
:+1: to all of the above. I'm happy to extensively test during development and also contribute code if needed.

**Comment by jmadin** [comment]
@caterinap @rachaelgallagher I started developing something in ruby on rails. There is a mock website up at Heroku [here](https://afternoon-tor-83256.herokuapp.com). The code is [here](https://github.com/jmadin/trait_registry) and could be brought over to this group.
 
It's very preliminary. I thought I should show it before putting any major time into it, given this discussion.  Just tried a few things out - and only a few of the registry fields added so far. A nice thing about the web app, is that there is a built in json / yaml APIs, which should make automating dataset merging etc easier for madDB.

The wiki idea is a good one. I've messed a bit with wiki-like things that interface with Github, like Jekyll. And so this might be worth looking into as well.

**Comment by rachaelgallagher** [comment]
hey @jmadin - thanks for getting this started on Ruby. I think it makes sense to group in this way (using major lineages) and maybe down the track we can do something fancy with an interactive phylogeny that will link you out to what trait initiatives are in progress/complete. 

whatever we can do to streamline the process of contributing to the registry will be excellent. I am in Tucson this week finalising the manuscript for submission (ideally in the next few weeks, but contingent on how rapidly i get approval from all co-authors).

**Comment by caterinap** [comment]
@jmadin that looks very nice! Should we link to this in the paper? I guess the only thing that would need to work is the possibility of adding new datasets (and maybe also have all registry fields). If you think that this is possible to implement in 1/2 weeks then we can add the link to the OpenTraits webpage. Otherwise I'll proceed with the temporary googleform solution (any help to make it properly work with github will be more than welcome!). @all: what do you think?

**Comment by jhpoelen** [comment]
Great to see such lively discussion and software experimentation! 

I think that existing infrastructures should be carefully considered before taking on the responsibility to design, implement, and maintain a new one. I am a fan of prototyping (e.g., https://en.wikipedia.org/wiki/Paper_prototyping !) to better understand needs and am also aware of the tremendous costs (e.g., time, money, attention) of developing a new piece of software from scratch. I hope that infrastructure re-use will help to keep the focus of the open traits network on helping to connect people and keep time spent on building custom software to a minimum. 

For the paper, I imagine that a reference to http://opentraits.org would suffice.

**Comment by jhpoelen** [comment]
@caterinap along these lines in my previous reply, I'd say that sticking with google forms coupled with a view-only google sheet would be sufficient for the time being.

**Comment by jmadin** [comment]
Sensible to stick with the google sheet.  Custom making a web app does need serious thought and discussion to do properly. Which could take along time. We definitely don't want to point to something half-baked. 

I'll move ahead with the prototype anyway, because it's a research interest of mine -- especially linking traits with existing controlled vocabularies / ontologies. If you're interested, I just seeded the prototype with the datasets from your trait standard paper, @caterinap

**Comment by caterinap** [comment]
I pushed a first .rmd file to the trait registry folder. It creates a table from the responses of the google form. As you can see the code is less than minimal and it's really just a start, so please feel free to change anything (and/or also reorganise the folder in a more efficient way, sorry about that!). @shandongfx and others working on the site: maybe you don't even need the .Rmd but just the [link ](https://docs.google.com/spreadsheets/d/e/2PACX-1vTGsH57fb-H1J-n3kCXrN_h6nCsZDd9TSXqXPxC0VEXZrzCpR0KjhoESJP_2JQHtU9OXRPtRx8e129Q/pub?output=csv) to download the responses. Oh, and the link to the [google form](https://docs.google.com/forms/d/e/1FAIpQLSdWL1hMzSGOfSSOGDFhjwipT1a1j9XSLpiDoI0ziTEMywsW7w/viewform?usp=sf_link) too.

**Comment by diatomsRcool** [comment]
There are several ontologies in the OBO Foundry that focus on phenotypes and traits. If we can't find the terms we need, let's get it added to the relevant ontology. I've done this before and I can help.

**Comment by jhpoelen** [comment]
@diatomsRcool excellent comment, I've opened a separate issue on that https://github.com/open-traits-network/open-traits-network.github.io/issues/144 . 

Am closing this issue, because the original issue discussed how to establish and manage opentraits.org  registries. We settled on using github / pull requests etc. for keep and share the member and data registries.
